src/query/storages/fuse/src/pruning/pruning_executor.rs

```diff
@@ -150,15 +151,33 @@ impl BlockPruner {
             .map_err(|e| ErrorCode::StorageOther(format!("block pruning failure, {}", e)))?;
 
         // 3. collect the result
-        tracing::debug_span!("collect_result").in_scope(|| {
-            // flatten the collected block metas
-            let metas = joint
-                .into_iter()
-                .collect::<Result<Vec<_>>>()?
-                .into_iter()
-                .flatten();
-            Ok(metas.collect())
-        })
+        let metas: Result<Vec<(usize, BlockMeta)>> = tracing::debug_span!("collect_result")
+            .in_scope(|| {
+                // flatten the collected block metas
+                let metas = joint
+                    .into_iter()
+                    .collect::<Result<Vec<_>>>()?
+                    .into_iter()
+                    .flatten();
+                Ok(metas.collect())
+            });
+        let metas = metas?;
+
+        // if there are ordering + limit clause, use topn pruner
```

Comment:
I push down the order_by and limit in 
[src/query/service/src/sql/optimizer/rule/rewrite/rule_push_down_sort_scan.rs](https://github.com/datafuselabs/databend/pull/7273/files#diff-8b117d9593a34e61fd76a5946e339731c0e5ac90530d380a6c664bb3c8292064)
```rust
    fn apply(&self, s_expr: &SExpr, state: &mut TransformState) -> Result<()> {
        let sort: Sort = s_expr.plan().clone().try_into()?;
        let child = s_expr.child(0)?;
        let mut get: LogicalGet = child.plan().clone().try_into()?;
        if get.order_by.is_none() {
            get.order_by = Some(sort.items);
        }
        if let Some(limit) = sort.limit {
            get.limit = Some(get.limit.map_or(limit, |c| cmp::max(c, limit)));
        }
        let get = SExpr::create_leaf(RelOperator::LogicalGet(get));
        state.add_result(s_expr.replace_children(vec![get]));
        Ok(())
    }
```

if only valid when ordering + limit clause, maybe I could modify the apply?

Only the in order by + limit case apply the partten push_down_sort_scan?

Reply:
It's ok now.

1. There is only a limit clause, it'll use LimitPruner (this is an accurate pruner).
2. There is only order clause, it'll not use LimitPruner and TopN Pruner.
3. There is `limit + order` clause, it'll use TopN Pruner.